File: Swift4/Swift4/model/Lokacija.swift
//
//  Lokacija.swift
//  Swift4
//
//  Created by Tomislav Jakopec on 22/02/2020.
//  Copyright © 2020 Tomislav Jakopec. All rights reserved.
//

import Foundation
class Lokacija: Entitet{
    private var longitude: Double
    private var latitude: Double
    private var naziv: String
    
     init(sifra: Int = 0, longitude: Double = 0, latitude: Double = 0, naziv: String) {
        self.longitude=longitude
        self.latitude=latitude
        self.naziv=naziv
        super.init(sifra: sifra)
    }
    
    public func getLongitude()->Double{
        return self.longitude
    }
    
    public func setLongitude(longitude: Double){
        self.longitude = longitude
    }
    
    public func getLatitudee()->Double{
        return self.latitude
    }
    
    public func setLatitude(latitude: Double){
        self.latitude = latitude
    }
    
    public func getNaziv()->String{
        return self.naziv
    }
    
    public func setNaziv(naziv: String){
        self.naziv = naziv
    }
    
}

File: JS/JS/lokacija.js
import { Entitet } from "./entitet.js"

export class Lokacija extends Entitet{
    #longitude
    #latitude
    #naziv

    constructor(id, longitude, latitude, naziv){
        super(id)
        this.#longitude = longitude
        this.#latitude = latitude
        this.#naziv = naziv
    }

    getLongitude(){
        return this.#longitude
    }
    setLongitude(longitude){
        this.#longitude = longitude
    }

    getLatitude(){
        return this.#latitude
    }
    setLatitude(latitude){
        this.#latitude = latitude
    }

    getNaziv(){
        return this.#naziv
    }
    setNaziv(naziv){
        this.#naziv = naziv
    }
}